### PR TITLE
ci: add chromatic github action that triggers on PR comment

### PR DESCRIPTION
**Pull request by mbylstra**
# Objective
- Add Chromatic github integration using the official Chromatic Github Action
- link it to the Culture Amp chromatic account (was previously linked to a personal account)
- The github action only publishes to Chromatic if the latest comment in the PR contains "run chromatic"
- The action should run whenever someone adds a comment(however, there is no way to test until we merge to master, as the `issue_comment` trigger is apparently only triggered for actions on the master branch)
- I tested this with the `push` trigger, which can be run on a dev branch (the final commit switches to `issue_comment`). This is an example of a succesful run that published to chromatic: https://github.com/cultureamp/kaizen-design-system/runs/2760935408?check_suite_focus=true

### TODO
- [x] create a Chromatic project for Kaizen in the Culture Amp Chromatic account
- [x] add a `chromatic.yaml` file to `.github/workflows` that runs the official Chromatic github action
- [x] make chromatic builds opt-in per commit (the default is to run the on every commit, which would quickly eat up all of our credits)
- [ ] delete the old buildkite chromatic step that was linked to a chromatic project on Michael's chromatic account.

### Notes:
-  the `issue_comment` is for both issue comments and PR comments (that's not a mistake, the Github Actions API is just really confusing!)
- I plan to the delete the old buildkite chromatic step in a follow up PR (just in case this one isn't working, there is still the ability to optionally use the old one)

# Motivation and Context
https://github.com/cultureamp/kaizen-design-system/issues/1602

### CODE REVIEW

**Comment by mbylstra** [comment]
This comment should kick of the chromatic action.

**Comment by mbylstra** [comment]
Maybe changing from `pull_requestion_review_comment` to `issue_comment` will work? Does this comment tigger a chromatic build?

**Comment by mbylstra** [comment]
run chromatic

**Comment by mbylstra** [comment]
this is a comment without the special command, so chromatic should not run on next push

File: .github/workflows/chromatic.yaml
Comment (mbylstra) [inline .github/workflows/chromatic.yaml:8]: this would be changed to:

```
on:
  issue_comment:
    types: [created]
 ```
 
 before merging, but the action can only be tested with `on: push`

File: .github/workflows/chromatic.yaml
Comment (mbylstra) [inline .github/workflows/chromatic.yaml:66]: I'm sure there's a way to turn this into a variable, but I don't know how!

File: .github/workflows/chromatic.yaml
Comment (mbylstra) [inline .github/workflows/chromatic.yaml:61]: I'm sure there's a way to turn this into a variable to avoid repetition, but I don't know how!

File: .github/workflows/chromatic.yaml
Comment (mbylstra) [inline .github/workflows/chromatic.yaml:41]: this is a hack, but it seems to work!

File: .github/workflows/chromatic.yaml
Comment (mbylstra) [inline .github/workflows/chromatic.yaml:46]: this was just some debug code that should be removed

**Comment by mbylstra** [comment]
run chromatic

**Comment by ActuallyACat** [comment]
I feel like we should update the documentation with instructions on how to run Chromatic now (potentially use the PR template)? https://github.com/cultureamp/kaizen-design-system/blob/master/.github/pull_request_template.md)

File: .github/workflows/chromatic.yaml
Comment (ActuallyACat) [inline .github/workflows/chromatic.yaml:47]: /nit Something like `@chromatic run` is a little more extensible and would allow us to write new command should we need them in the future. I like the way Dependabot has approached it: 
```
@dependabot rebase will rebase this PR
@dependabot recreate will recreate this PR, overwriting any edits that have been made to it
@dependabot merge will merge this PR after your CI passes on it
... etc
```

File: .github/workflows/chromatic.yaml
Comment (ActuallyACat) [inline .github/workflows/chromatic.yaml:56]: I might be misunderstanding something, but why do we need to store this information (`key`)?

**Review by ActuallyACat — CHANGES_REQUESTED** [review]
Just have a few small suggestions

File: .github/workflows/chromatic.yaml
Comment (mbylstra) [reply .github/workflows/chromatic.yaml:56]: I'm not really sure, that was copy pasta from the unified-home config.

File: .github/workflows/chromatic.yaml
Comment (mbylstra) [reply .github/workflows/chromatic.yaml:47]: I wasn't sure about that, because `@dependabot` actually at-mentions the dependabot account/user. I think @chromatic would at-mention chromatic, and they'd get a bunch of notifications. We could do `@cultureamp`, but I wasn't sure if that would also generate a bunch of notifications we don't want.

**Comment by mbylstra** [comment]
>I feel like we should update the documentation with instructions on how to run Chromatic now (potentially use the PR template)? https://github.com/cultureamp/kaizen-design-system/blob/master/.github/pull_request_template.md) 

Ah yeah, I was intending to do that. I can add that to the follow up PR where the old chromatic buildkite step gets removed. I'm not 100% sure this works yet, and it can only be properly tested once merged, so I didn't want to add the instructions until we knew it works properly.

File: .github/workflows/chromatic.yaml
Comment (chromatic) [reply .github/workflows/chromatic.yaml:47]: > I think @chromatic would at-mention chromatic (this person: https://github.com/chromatic), and they'd get a bunch of notifications.

Confirmed! No harm done though.

Comment (ActuallyACat) [reply .github/workflows/chromatic.yaml:47]: Yeah, good call (and thanks for chiming in @chromatic 😄). We also have a @cultureamp-ci account that we can use https://github.com/cultureamp-ci. @jasononeil or @zioroboco might have some thoughts on whether a PR bot for common actions is something we might want to invest time in and make available for other repos.

Comment (ActuallyACat) [reply .github/workflows/chromatic.yaml:47]: (That being said, I don't think this conversation should block this PR and would be happy for this to be merged with `chromatic run` as the trigger)

File: .github/workflows/chromatic.yaml
Comment (ActuallyACat) [reply .github/workflows/chromatic.yaml:56]: Ah! Okay looking at unified's config and these docs it looks like a pretty standard way to cache deps: https://docs.github.com/en/actions/guides/caching-dependencies-to-speed-up-workflows

**Review by ActuallyACat — APPROVED** [review]
LGTM thanks @mbylstra 👍🏻

**Comment by mbylstra** [comment]
Thanks for the review @ActuallyACat. I might hold off on merging until next Monday, as I think I'll be too stretched to follow it up this week if there are issues. I'm totally happy for you to move it forward if you have the time though.

**Comment by ActuallyACat** [comment]
Good call.

**Pull request merged**
